blockly-core/core/ui/contract_editor/contract_definition_section.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+
+goog.provide('Blockly.ContractDefinitionSection');
+goog.require('Blockly.BlockSpaceEditor');
+
+/**
+ * Margin to leave above the function block below the flyout in pixels.
+ * @type {number}
+ * @const
+ */
+var FUNCTION_BLOCK_VERTICAL_MARGIN = Blockly.BlockSpaceEditor.BUMP_PADDING_TOP;
+
+/**
+ * Handles UI state and layout for the contract editor definition section/block
+ * @param {Element} canvasToDrawOn element this section should be drawn on
+ * @constructor
+ */
+Blockly.ContractDefinitionSection = function (canvasToDrawOn) {
+  this.definitionTableGroup = Blockly.createSvgElement('g', {}, canvasToDrawOn);
+  this.grayDefinitionBackground = Blockly.createSvgElement('rect', {
+    'fill': '#DDD'
+  }, this.definitionTableGroup);
+
+  this.verticalDefinitionMidline = Blockly.createSvgElement('rect', {
+    'fill': '#000'
+  }, this.definitionTableGroup);
+  this.verticalDefinitionMidline.setAttribute('width', 2.0);
+
+  this.horizontalDefinitionTopLine = Blockly.createSvgElement('rect', {
+    'fill': '#000'
+  }, this.definitionTableGroup);
+  this.horizontalDefinitionTopLine.setAttribute('height', 2.0);
+  this.horizontalDefinitionBottomLine = Blockly.createSvgElement('rect', {
+    'fill': '#000'
+  }, this.definitionTableGroup);
+  this.horizontalDefinitionBottomLine.setAttribute('height', 2.0);
+};
+
+/**
+ * Update UI to match new collapse state
+ * @param {boolean} isNowCollapsed - whether we are now collapsed
+ */
+Blockly.ContractDefinitionSection.prototype.handleCollapse = function (isNowCollapsed) {
+  this.definitionTableGroup.style.display = isNowCollapsed ? 'none' : 'block';
```

Comment:
This also used to include the two lines:

```
this.verticalDefinitionMidline.style.display = isNowCollapsed ? 'none' : 'block';
this.grayDefinitionBackground.style.display = isNowCollapsed ? 'none' : 'block';
```

Since the `this.definitionTableGroup` is the parent of those two, that appears to be sufficient. @Bjvanminnen I was a little superstitiously also hiding the children, but we don't know of any reason hiding a parent group wouldn't hide its children, right?

I can also take a quick peek in IE9/10, Firefox and Safari just to be sure